server/test/unit/server/webservices/views/test_repositories.py

```diff
@@ -0,0 +1,1977 @@
+        mock_collection.return_value.find.return_value = mock_repos
+        mock_request = mock.MagicMock()
+        mock_request.GET = {'distributors': 'true'}
+        repos_view = ReposView()
+
+        repos_view.get(mock_request)
+        mock_process.assert_called_once_with(mock_repos, False, True)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_CREATE())
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.repositories.generate_redirect_response')
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_post_repos_only_id(self, mock_factory, mock_reverse, mock_redir, mock_resp):
+        """
+        Create a repo using the minimal body.
+        """
+
+        mock_request = mock.MagicMock()
+        mock_request.body = json.dumps({'id': 'mock_repo'})
+        mock_reverse.return_value = '/mock/path/'
+        mock_repo_manager = mock_factory.repo_manager.return_value
+        mock_repo_manager.create_and_configure_repo.return_value = {'id': 'mock_repo'}
+
+        repos_view = ReposView()
+        response = repos_view.post(mock_request)
+
+        expected_content = {'id': 'mock_repo', '_href': '/mock/path/'}
+        mock_resp.assert_called_once_with(expected_content)
+        mock_redir.assert_called_once_with(mock_resp.return_value, expected_content['_href'])
+        self.assertTrue(response is mock_redir.return_value)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_CREATE())
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.repositories.generate_redirect_response')
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_post_repos_all_fields(self, mock_factory, mock_reverse, mock_redir, mock_resp):
+        """
+        Create a repo using all allowed fields.
+        """
+
+        mock_repo = {
+            'id': 'mock_repo', 'display_name': 'mock_display', 'description': 'mock_desciption',
+            'notes': 'mock_notes', 'importer_type_id': 'mock_importer', 'distributors': ['dist1']
+        }
+        mock_request = mock.MagicMock()
+        mock_request.body = json.dumps(mock_repo)
+        mock_reverse.return_value = '/mock/path/'
+        mock_repo_manager = mock_factory.repo_manager.return_value
+        mock_repo_manager.create_and_configure_repo.return_value = mock_repo
+
+        repos_view = ReposView()
+        response = repos_view.post(mock_request)
+
+        mock_repo['_href'] = '/mock/path/'
+        mock_resp.assert_called_once_with(mock_repo)
```

Comment:
in my opinion it is better to use three vars in this case mock_repo for the request body, then mock_repo1 for the return_value of mock_repo_manager and mock_repo2 when you do the assertion here - mock_resp.assert_called_once_with.
reason why it is better - because usually the response body is different from request body,plus you will be sure that mock_repo content will not be changed somewhere in the middle if you forgot something to mock